### PR TITLE
fix release script

### PR DESCRIPTION
**Pull request by TkDodo**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,7 +25,6 @@ jobs:
           cache: 'npm'
       - run: |
           npm i
-          npm run test:ci
```
Comment (TkDodo) [inline .github/workflows/ci.yml:28]: not needed because the publishci script also runs test

File: scripts/publish.ts [hunk 2]
```diff
@@ -167,7 +167,7 @@ async function run() {
     ? packages
     : changedFiles.reduce((changedPackages, file) => {
         const pkg = packages.find((p) =>
-          file.startsWith(path.join('packages', p.packageDir, p.srcDir)),
+          file.startsWith(path.join('packages', p.packageDir)),
```
Comment (TkDodo) [inline scripts/publish.ts:170]: makes sure that we publish new versions whenever there are changes anywhere in the package directory, not just the `src` directory!

File: scripts/publish.ts [hunk 3]
```diff
@@ -177,16 +177,28 @@ async function run() {
...
+    const allDependencies = Object.keys(
+      Object.assign(
+        {},
+        packageJson.dependencies ?? {},
+        packageJson.peerDependencies ?? {},
+      ),
+    )
+
     if (
-      pkg.dependencies?.find((dep) =>
```
Comment (TkDodo) [inline scripts/publish.ts:182]: `pkg.dependencies` does not exist - the type were wrong
so we need to find them in packgage.json

File: scripts/publish.ts [hunk 4]
```diff
@@ -419,59 +427,55 @@ async function run() {
       path.resolve(rootDir, 'packages', pkg.packageDir, 'package.json'),
       async (config) => {
         await Promise.all(
-          (pkg.dependencies ?? []).map(async (dep) => {
+          Object.keys(config.dependencies ?? {}).map(async (dep) => {
```
Comment (TkDodo) [inline scripts/publish.ts:430]: same here - pkg.dependencies does not exist

File: scripts/publish.ts [hunk 5]
```diff
@@ -419,59 +427,55 @@ async function run() {
...
             const depPackage = packages.find((d) => d.name === dep)
 
-            if (!depPackage) {
-              throw new Error(`Could not find package ${dep}`)
```
Comment (TkDodo) [inline scripts/publish.ts:426]: throwing here is wrong as it fails the promise and `Promise.all` stops after the first rejection. We can just ignore dependencies that are _not_ dependencies we care about

**Review by DamianOsipiuk — APPROVED** [review]
Nice 🥇

**Pull request merged**